### DISCUSSION TITLE
Using SlintPad with Vivi?

### DISCUSSION THREAD

**Discussion by npwoods** [slint-ui/slint#5620] [Q&A]
_(please_ forgive me if this is not the best place to ask this question)_

I was looking at [vivi](https://app.radicle.xyz/nodes/seed.radicle.garden/rad:z3oxAZSLcyXgpa7fcvgtueF49jHpH), which promises to be a good add on to Slint.  I was able to get it to work locally, but I was curious how I would use it with Slintpad?

**Comment by hunger** [comment]
That's a very good question. I need to figure out whether I can enable that in slintpad somehow.

Rust playground just imports the top x% of all crates, I wonder whether we can do something similar in Slintpad.

**Comment by ghost** [comment]
The `@ ` imports currently don't work with SintPad. If you want to use it for your own stuff instead using the `@` imports you can copy the vivi slint files relative to your slint files and use path based imports then instead:

```slint
import { FilledButton } from "../path/to/vivi/magic.slint";
```

**Comment by ogoffart** [comment]
We should add some way to do it in the URL at least, and maybe even through some config UI.

**Reply by ogoffart** [reply]
This was just added in slintpad, by adding a `?lib=vivi=...` in the URL

Here is an example URL from the snapshot (it will only be available in slintpad when version 1.14 is released)

https://snapshots.slint.dev/master/editor/?lib=vivi=https://seed.radicle.garden/raw/rad:z3oxAZSLcyXgpa7fcvgtueF49jHpH/92a0987cf92647290353826bf05113a65fca25a9/vivi/ui&gz=H4sIAAAAAAAAA1WMQQqDQAxF9znFZw6g-3YjIuI1ik0lkJmITUWQuXtn2pVv-eA9iattjhOjqPKz_7hbQsZrs4jQ7bJLGx-LzM1bJXm4E_HxS2YraeLkGDgaTkLhcvmrivPhN4SJVQ31WT7VZ8r0BVEf-8qCAAAA
